src/channel.js

```diff
@@ -409,6 +409,15 @@ export class Channel {
 		this.initialized = true;
 		this._initializeState(state);
 		this.data = state.channel;
+
+		this._client.logger(
+			'info',
+			'channel:watch() - started watching channel ' + this.cid,
+			{
+				tags: ['channel'],
+				channel: this,
+			},
+		);
```

Comment:
we should attach events to logging as well ie.

```js
client.on((e) => {client.logger(...)})
```

Reply:
I have added a log in`.on(...)` now. Although it only attaches the listeners. When event is received,  `_handleChannelEvent ` and `_handleClientEvent ` are the functions that get called. I have already added logs to those functions :)